engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -287,26 +287,23 @@ def scc_activate_product
           auth = nil
           auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
           mode = nil
+          token_not_provided = !params.key?(:token) || params[:token].empty?
           if @system.byos?
             mode = 'byos'
           elsif !@product.free? && @product.extension?
-            regcode_error_message = 'A registration code is required to activate this product'
-            raise ActionController::TranslatedError.new(regcode_error_message) if (@system.payg? || @system.hybrid?) && params[:token].nil?
-
-            # system is not BYOS and
-            # product is a non free extension
-            # and a registration code is provided
-            mode = 'hybrid'
-            # the extensions must be the same version and arch
-            # than base product
-            base_prod = @system.products.find_by(product_type: :base)
-            if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
-              request.headers['proxy_byos_mode'] = mode
-              request.headers['scc_login'] = @system.login
-              request.headers['scc_password'] = @system.password
-              request.headers['instance_data'] = params[:instance_data]
-              request.headers['hwinfo'] = params[:hwinfo]
-              response = SccProxy.announce_system_scc(auth, request.headers)
+            unless token_not_provided
+              mode = 'hybrid'
+              # the extensions must be the same version and arch
+              # than base product
+              base_prod = @system.products.find_by(product_type: :base)
+              if @system.payg? && base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
+                request.headers['proxy_byos_mode'] = mode
+                request.headers['scc_login'] = @system.login
+                request.headers['scc_password'] = @system.password
+                request.headers['instance_data'] = params[:instance_data]
+                request.headers['hwinfo'] = params[:hwinfo]
+                response = SccProxy.announce_system_scc(auth, request.headers)
```

Comment:
would [4ad0c34](https://github.com/SUSE/rmt/pull/1203/commits/4ad0c3442213506fe0e4fe61ea89390ba93a588e) + [this](https://github.com/SUSE/rmt/blob/master/engines/scc_proxy/lib/scc_proxy/engine.rb#L95) be enough ? I think it would

Reply:
I don't think so. We're intercepting an activation call here, and sending an announce call to SCC. They have different auth methods, that's why we cannot copy the `request.headers['HTTP_AUTHORIZATION']` into `auth`. As described above, the `auth` needs to get build with the regcode. 
Also, I don't know if `params[:instance_data]` and `params[:hwinfo]` are actually part of the client's request payload. Maybe those values need to get taken from `@system` instead.

Reply:
@digitaltom I think that is addressed in [75ee5b0](https://github.com/SUSE/rmt/pull/1203/commits/75ee5b0e977cb5d987a7e1b23e55071e24d10ac5)